node/index.js

```diff
@@ -11,87 +11,151 @@ var plaid = require('plaid');
 var PUBLIC_TOKEN = null;
 var ITEM_ID = null;
-// The payment_token is only relevant for the UK Payment Initiation product.
-// We store the payment_token in memory - in production, store it in a secure
+// The payment_id is only relevant for the UK Payment Initiation product.
+// We store the payment_id in memory - in production, store it in a secure
 // persistent data store
-var PAYMENT_TOKEN = null;
 var PAYMENT_ID = null;
 
 // Initialize the Plaid client
 // Find your API keys in the Dashboard (https://dashboard.plaid.com/account/keys)
-var client = new plaid.Client(
-  PLAID_CLIENT_ID,
-  PLAID_SECRET,
-  PLAID_PUBLIC_KEY,
-  plaid.environments[PLAID_ENV],
-  {version: '2019-05-29', clientApp: 'Plaid Quickstart'}
-);
+var client = new plaid.Client({
+  clientID: PLAID_CLIENT_ID,
+  secret: PLAID_SECRET,
+  env: plaid.environments[PLAID_ENV],
+  options: {
+    version: '2019-05-29',
+  }
+});
 
 var app = express();
 app.use(express.static('public'));
-app.set('view engine', 'ejs');
 app.use(bodyParser.urlencoded({
   extended: false
 }));
 app.use(bodyParser.json());
 
 app.get('/', function(request, response, next) {
-  response.render('index.ejs', {
-    PLAID_PUBLIC_KEY: PLAID_PUBLIC_KEY,
-    PLAID_ENV: PLAID_ENV,
-    PLAID_PRODUCTS: PLAID_PRODUCTS,
-    PLAID_COUNTRY_CODES: PLAID_COUNTRY_CODES,
-    PLAID_OAUTH_REDIRECT_URI: PLAID_OAUTH_REDIRECT_URI,
-    PLAID_OAUTH_NONCE: PLAID_OAUTH_NONCE,
-    ITEM_ID: ITEM_ID,
-    ACCESS_TOKEN: ACCESS_TOKEN,
-  });
+  response.sendFile('./views/index.html', { root: __dirname });
 });
 
 // This is an endpoint defined for the OAuth flow to redirect to.
 app.get('/oauth-response.html', function(request, response, next) {
-  response.render('oauth-response.ejs', {
-    PLAID_PUBLIC_KEY: PLAID_PUBLIC_KEY,
-    PLAID_ENV: PLAID_ENV,
-    PLAID_PRODUCTS: PLAID_PRODUCTS,
-    PLAID_COUNTRY_CODES: PLAID_COUNTRY_CODES,
-    PLAID_OAUTH_NONCE: PLAID_OAUTH_NONCE,
-  });
+  response.sendFile('./views/oauth-response.html', { root: __dirname });
+});
+
+app.post('/api/info', function(request, response, next) {
+  response.json({
+    item_id: ITEM_ID,
+    access_token: ACCESS_TOKEN,
+    products: PLAID_PRODUCTS
+  })
+});
+
+app.post('/api/create_link_token', function(request, response, next) {
+  client.createLinkToken(
+    {
+       'user': {
+         // This should correspond to a unique id for the current user.
+         'client_user_id': 'user-id',
+       },
+       'client_name': "Plaid Quickstart",
+       'products': PLAID_PRODUCTS,
+       'country_codes': PLAID_COUNTRY_CODES,
+       'language': "en",
+       'redirect_uri': PLAID_REDIRECT_URI,
+     }
+    , function(error, createTokenResponse) {
+      if (error != null) {
+        prettyPrintResponse(error);
+        return response.json({
+          error: error,
+        });
+      }
+      response.json(createTokenResponse);
+  })
+});
+
+app.post('/api/create_link_token_for_payment', function(request, response, next) {
```

Comment:
nit: Seems like the other routes have links to the docs where relevant. We can add https://plaid.com/docs#payment-initiation-create-link-token-request